Odata-docs/odatalib/whats-new/odatalib-8.md

```diff
@@ -0,0 +1,429 @@
+    "UserName": "scottketchum",
+    "FirstName": "Scott",
+    "LastName": "Ketchum",
+    ...
+}
+```
+
+However, in OData .NET 8, this code would generate the following URL: `https://services.odata.org/TripPinRESTierService/People?$filter=UserName eq 'russellwhyte'&$expand=BestFriend`. Notice that request filters the entity set collection and also includes the navigation property using the `$expand` query. This means the result will be a collection containing a single entity, which has an expanded `BestFriend` property:
+
+```csharp
+{
+    "@odata.context": "https://services.odata.org/TripPinRESTierService/(S(pbicrumqvat4igher2mbadb3))/$metadata#People(BestFriend())",
+    "value": [
+        {
+            "UserName": "russellwhyte",
+            "FirstName": "Russell",
+            "LastName": "Whyte",
+            ...
+            "BestFriend": {
+                "UserName": "scottketchum",
+                "FirstName": "Scott",
+                "LastName": "Ketchum",
+                ...
+            }
+        }
+    ]
+}
+```
+
+This means that trying to access the result with `friend.FirstName` will fail and throw an exception, because the result is not the friend entity, but a collection that contains the filtered entity with the `BestFriend` property attached to it. Therefore in OData .NET 8 we would have to adjust the code as follows:
+
+```csharp
+// fetch people with expanded navigation property using OData URI People?$expand=BestFriend
+var people = await context.People.Expand('BestFriend').ExecuteAsync();
+// perform a local LINQ transform on the result data
+var bestFriends = people.Select(p => p.BestFriend);
+
+Console.WriteLine(bestFriends.First().FirstName);
+```
+
+Alternatively we could use the `ByKey` method:
+
+```csharp
+var friend = await ctx.People.ByKey("russelwhyte").Select(p => p.BestFriend).GetValueAsync();
+Console.WriteLine(friend.FirstName);
+```
+
+Alternatively, we could set `DataServiceContext.KeyComparisonGeneratesFilterQuery` to false, but it's no longer recommended:
+
+```csharp
+var context = new Container(new Uri("https://services.odata.org/TripPinRESTierService/"));
+context.KeyComparisonGeneratesFilterQuery = false;
+
+var friend = context.People.Where(p => p.UserName == "russellwhyte").Select(p => p.BestFriend);
+
+Console.WriteLine(friend.FirstName);
+```
+
+### Dropped support for `HttpWebRequest`
+
+OData .NET 8 removed support for `HttpWebRequest`, `HttpClient` should be used instead.
+
+Subsequently the following related APIs have also been removed:
+
+- The `HttpWebRequestMessage` class was removed.
+- The `HttpRequestTransportMode` enum was removed.
+- The `DataServiceContext.HttpRequestTransportMode` property was removed.
+
+### Other obsolete APIs that have been removed
+
+- Obsolete `Credentials` property dropped from `DataServiceClientRequestMessage` abstract class. The recommended way to configure credentials is through `HttpClientHandler` that can be provided using `IHttpClientFactory`.
```

Comment:
Is this necessary to provide a small snippet of how to configure this?

Reply:
I've added code samples in the section that talks about `IHttpClientFactory` support.